packages/embed/src/index.ts

```diff
@@ -175,11 +170,23 @@ export function setup(setupConfig: SetupConfig) {
     appleAbortSession: subjectManager.appleAbortSession$.next,
     googlePaySessionStarted: subjectManager.googlePaySessionStarted$.next,
     googlePaySessionCompleted: subjectManager.googlePaySessionCompleted$.next,
-    frameReady: ({ version } = {}) => {
+    frameReady: async ({ version } = {}) => {
       if (version) {
         setVersion('embed-ui', version)
       }
       clearTimeout(frameLoadWarn)
+
+      // Apple Pay
+      await hasLoadedApplePaySdk
```

Comment:
We're not actually using the resolved value of this right? Should we to prevent calling `browserSupportsApplePay` when it resolves to `false`?

Reply:
We still need to run `browserSupportsApplePay` in the Safari scenario, where `loadApplePaySdk` resolves to false. We're not using the returned Promise value, but we wait for it to be done (load the script or time out) before running `browserSupportApplePay`. Not only it does check `window.ApplePaySession` again, but also detects if the device can make payments and supports the version

Reply:
right, I supposed we could do:

```
const supportedApplePayVersion = hasLoadedApplePaySdk && browserSupportsApplePay()
        ? detectSupportedVersion()
        : 0
```

though it wouldn't do much as optimization